roomserver/internal/helpers/helpers.go

```diff
@@ -50,6 +50,10 @@ func UpdateToInviteMembership(
 	return updates, nil
 }
 
+// IsServerCurrentlyInRoom checks if a server is in a given room, based on the room
+// memberships. If the servername is not supplied then the local server will be
+// checked instead using a faster code path.
```

Comment:
It feels like a bug to call `IsServerCurrentlyInRoom` without.. a server? Why have this weird API shape?

Reply:
Mostly because nothing in `IsServerCurrentlyInRoom` knows the current server name to compare against, and then it also has the same behaviour as `QueryServerJoinedToRoomRequest`.

Reply:
Bleh, I don't like it. I guess we can fix that later.